packages/common/src/core/schemas/internal/EditorOptions.ts

```diff
@@ -0,0 +1,19 @@
+import { HubEntity } from "../../types";
+
+/** Intersection type of all EditorOptions */
+export type EditorOptions = IEntityEditorOptions | ICardEditorOptions;
```

Comment:
I wonder about using a union type here. As you saw, in `getEditorConfig(...)` fn, you had to use `as ...` or typescript would complain. And really, what options structure is valid, depends on the `type` that's passed into `getEditorConfig(...)`... as in it's invalid to pass `type: "card"` and `options: IEntityEditorOptions` but this form of typing does not actually prevent that. 
One way to get around this is to use a [Discriminated Unions](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) which would combine the type string + the options props, and then you can be sure you're not getting mixed values that result in invalid conditions